### PR TITLE
Built server framework and connected to DB

### PR DESCRIPTION
**Pull request by pjol**
[Ticket](https://trello.com/c/ouZJohnF/24-m-build-server-framework-and-connect-to-db)

- Created server framework
- Implemented DB connection
- Created proof-of-life route

### CODE REVIEW

File: router.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    default_price: 1,
+    features: 1,
+  })
+  .then((data) => {
+    data.id = data.product_id;
+    res.send(data);
+  })
+})
+
+module.exports = router;
```
Comment (justcord10) [inline router.js:30]: just testing to see that you get one from the db and it works

File: router.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    name: 1,
+    slogan: 1,
+    description: 1,
+    category: 1,
+    default_price: 1,
+    features: 1,
+  })
+  .then((data) => {
+    data.id = data.product_id;
+    res.send(data);
```
Comment (justcord10) [inline router.js:23]: no catch block?

File: server.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+app.use('/', router);
+app.use(express.json());
+app.use(express.static(path.join(__dirname, './public/dist')));
+
+mongoose.connect(url)
+.then(() => {
+  app.listen(port, () => {
+    console.log('Products Server Live on Port ' + port);
+  })
```
Comment (justcord10) [inline server.js:22]: oh interesting that you connect to your database and then start the server, but you also want a catch block here for any errors

**Review by justcord10 — CHANGES_REQUESTED** [review]
looks good overall, i would just add a couple of catch blocks to your code to handle any errors where i left comments

File: router.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
+const router = require('express').Router();
+const path = require('path');
```
Comment (LukeLarson2) [inline router.js:2]: Is this needed in your router file? I scanned for it a couple times but I didnt see it, usually you would only need this in your server

**Review by LukeLarson2 — APPROVED** [review]
Nice work making the changes, making big moves

**Review by justcord10 — APPROVED** [review]
fixed what you said you were going to fix, good job!

**Pull request merged**